shared/actions/chat2/index.js

```diff
@@ -1183,36 +1183,51 @@ const messageSend = (action: Chat2Gen.MessageSendPayload, state: TypedState) =>
   const ephemeralData = ephemeralLifetime !== 0 ? {ephemeralLifetime} : {}
 
   // Inject pending message and make the call
-  return Saga.sequentially([
-    Saga.put(
-      Chat2Gen.createMessagesAdd({
-        context: {type: 'sent'},
-        messages: [
-          Constants.makePendingTextMessage(
-            state,
-            conversationIDKey,
-            text,
-            Types.stringToOutboxID(outboxID.toString('hex') || ''), // never null but makes flow happy
-            ephemeralLifetime
-          ),
-        ],
-      })
-    ),
-    Saga.call(
-      RPCChatTypes.localPostTextNonblockRpcPromise,
-      {
-        ...ephemeralData,
-        body: text.stringValue(),
-        clientPrev,
-        conversationID: Types.keyToConversationID(conversationIDKey),
-        identifyBehavior: getIdentifyBehavior(state, conversationIDKey),
-        outboxID,
-        tlfName,
-        tlfPublic: false,
-      },
-      Constants.waitingKeyPost
-    ),
-  ])
+
+  const addMessage = Saga.put(
+    Chat2Gen.createMessagesAdd({
+      context: {type: 'sent'},
+      messages: [
+        Constants.makePendingTextMessage(
+          state,
+          conversationIDKey,
+          text,
+          Types.stringToOutboxID(outboxID.toString('hex') || ''), // never null but makes flow happy
+          ephemeralLifetime
+        ),
+      ],
+    })
+  )
+
+  const postText = Saga.call(
+    RPCChatTypes.localPostTextNonblockRpcPromise,
+    {
+      ...ephemeralData,
+      body: text.stringValue(),
+      clientPrev,
+      conversationID: Types.keyToConversationID(conversationIDKey),
+      identifyBehavior: getIdentifyBehavior(state, conversationIDKey),
+      outboxID,
+      tlfName,
+      tlfPublic: false,
+    },
+    Constants.waitingKeyPost
+  )
```

Comment:
Why did you factor these out into their own constants? They aren't used anywhere except in the `Saga.sequentially`.

Reply:
A couple of reasons:

1) The logs show that stuff stops after the messageSend action is sent. There's a slight chance that something in the constant builders above is causing an exception or something. So putting the log after all the constants have been built will narrow down where the stoppage is actually happening.

2) I was going to log some stuff from the constants, but couldn't find anything useful that wasn't also private. But I may want to in the future.

Reply:
For 1, unless you yield on the result of the `Saga.put`, aren't you racing with it?

For 2, it might be a good idea to add a comment in the code so nobody factors it back into the `Saga.sequentially`.

Reply:
`Saga.put` just makes an object. It doesn't actually do anything until you return it. Okay, I'll add a comment.

Reply:
(see https://github.com/redux-saga/redux-saga/blob/2e9a00cf84d5aa0b853f34790ccab89eedf8fe95/packages/core/src/internal/io.js#L55 )

Reply:
Don't understand. This should change nothing. Saga call also just makes an object so this just pulls two parameters out into variables and passes them in.  functionally nothing has changed

Reply:
i want to log something after the objects have been created and before they've been returned. it's possible that the object creation throws somehow (although I don't see how) and i want to rule that out.

Reply:
It's not meant to change anything functionally.

Reply:
I think the same thing will logically happen. If it throws it throws in the same place. The params are created in the same way and timing when passes to sequentially
Aka this is like

F(1+2,3+4) vs
A=1+2
B=3+4
F(A,B)